node/src/components/deploy_buffer.rs

```diff
@@ -290,57 +326,64 @@ where
     ) -> Effects<Self::Event> {
         match (&self.status, event) {
             (ComponentStatus::Fatal(msg), _) => {
-                error!(
+                warn!(
                     msg,
                     "should not handle this event when this component has fatal error"
                 );
                 Effects::new()
             }
-            (ComponentStatus::Uninitialized, Event::Initialize(blocks)) => {
-                for block in blocks {
-                    self.register_block_finalized(&block.into());
+            (ComponentStatus::Uninitialized, evt) => {
+                if let Event::Initialize(blocks) = evt {
+                    for block in blocks {
+                        self.register_block_finalized(&block.into());
+                    }
+                    self.status = ComponentStatus::Initialized;
+                    // start self-expiry management on initialization
+                    effect_builder
+                        .set_timeout(self.cfg.expiry_check_interval().into())
+                        .event(move |_| Event::Expire)
+                } else {
+                    warn!("should not handle this event when component is uninitialized");
```

Comment:
```suggestion
                    warn!(?evt, "should not handle this event when component is uninitialized");
```

(Might be useful in case it happens.)